logger/clickhouse/clickhouse_decorator.cc

```diff
@@ -0,0 +1,119 @@
+/**
+ *		Tempesta FW
+ *
+ * Copyright (C) 2024-2025 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+#include <fmt/format.h>
+#include <spdlog/spdlog.h>
+
+#include "../../libtus/error.hh"
+#include "clickhouse_decorator.hh"
+
+ClickHouseDecorator::ClickHouseDecorator(std::unique_ptr<IClickhouse> client,
+			    std::string_view table_template,
+			    std::string_view table_name,
+			    std::span<const TfwField> fields,
+			    size_t max_events)
+	: table_name_(table_name)
+	, table_creation_query_(fmt::format(fmt::runtime(table_template), table_name_))
+	, max_events_(max_events)
+	, client_(std::move(client))
+	, block_(make_block(fields, max_events))
+{
+	ensure_table_created();
+}
+
+ClickHouseDecorator::~ClickHouseDecorator()
+{
+	if (!flush(true))
+		handle_block_error();
+}
+
+bool
+ClickHouseDecorator::ensure_table_created() noexcept
+{
+	if (!needs_create_table_) [[likely]]
+		return true;
+
+	needs_create_table_ = client_->execute(table_creation_query_);
```

Comment:
```suggestion
	needs_create_table_ = !client_->execute(table_creation_query_);
```
We need to create table only if `execute` returned `false`, isn't?